### PR TITLE
Tmedia 264 localize date passed in value

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Pass in any locale

## Jira Ticket
- [TMEDIA-264](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-264&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
A client can specify a currently-unsupported dateLocalization.language (eg pt_PT).  Currently, for example, "EN" maps to "en_US" (American "Simplified" English). 

Any of the existing dateLocalization.language mappings (e.g. how we map “en” to “en_US”) should still be in effect 

If a customer adds a new value for dateLocalization.language that we don’t have mapped to something, then we should use that value directly 

Examples: en_GB should use en_GB

## Test Steps
- See doc debugging postinstall

// To test this, see debugging-timezone-postinstall-dates.md
// 1. Add a blocks.json file in the src/ folder from above doc
// 2. run `rm -rf node_modules && npm ci --production`
// 3. run `ls node_modules/timezone`
// 4. See pt_PT in the list
// 5. You can uncomment this test

// eslint-disable-next-line max-len
// it('support portuguese in portugal language and portugal lisbon timezone when setup with blocks.json', () => {
//   expect(localizeDateHelper(
//     '2000-01-02 01:00',
//     '%B %d, %Y %l:%M %P %Z',
//     'pt_PT',
//     'Europe/Lisbon',
//   )).toMatchInlineSnapshot('"Janeiro 02, 2000  1:00  WET"');
// });

### Before
- Portuguese and other locales not allowed

### After
- Can pass in any locale

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps above are working
- [x] Confirmed there are no linter errors
- [x] Confirmed this PR has reasonable code coverage
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added. -> yas

### CODE REVIEW

File: src/utils/localizeDate.test.ts [hunk 1]
```diff
@@ -26,8 +26,8 @@ it('returns American english language and utc date for timezone not found', () =
   ).toMatchInlineSnapshot('"January 02, 2000  1:00 am UTC"');
 });
 
-// supported locale
-// unsupported timezone
+// supported locale if no blocks.json found
+// unsupported timezone if no blocks.json found
 it('returns correct Korean language for locale ko but falls back to UTC', () => {
```
Comment (nschubach) [inline src/utils/localizeDate.test.ts:31]: This test is failing for me.

Comment (nschubach) [reply src/utils/localizeDate.test.ts:31]: image

**Review by nschubach — APPROVED** [review]
I don't know what the timezone library is doing in this case.  the test looks like it's trying to use the 'Asia/Seoul' timezone, which looks like the correct return value, but the test is expecting UTC?

Maybe it's a different version or an update.  Either way, if I comment this out, the other tests pass and the work here looks appropriate.  I assume it works for you since you added the Portuguese test.

File: src/utils/localizeDate.test.ts (hunk 1 above)
Comment (JackHowa) [reply src/utils/localizeDate.test.ts:31]: what did you run to see this? `rm -rf node_modules && npm i && npm t`? 

Did you have a `src/blocks.json`? where did you get one if so?

Comment (JackHowa) [reply src/utils/localizeDate.test.ts:31]: ah nvm found the issue; it's from my sample blocks.json in the doc. thank you! fixing

Comment (JackHowa) [reply src/utils/localizeDate.test.ts:31]: updated -- thank you @nschubach

**Comment by JackHowa** [comment]
also for testing live I used this https://github.com/WPMedia/Fusion-News-Theme/pull/181 for QA to do this

File: src/utils/localizeDate.test.ts (hunk 1 above)
Comment (nschubach) [reply src/utils/localizeDate.test.ts:31]: Sorry, I didn't see this until now.  Glad you found the issue!  It's interesting that setting it to New York made it result in UTC vs. Korea.

**Pull request merged**